kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -85,26 +122,51 @@ public ErrorStatusUpdateControl<KafkaProxy> updateErrorStatus(KafkaProxy primary
     }
 
     private static KafkaProxy buildStatus(KafkaProxy primary,
+                                          Context<KafkaProxy> context,
                                           @Nullable Exception exception) {
+        if (exception instanceof AggregatedOperatorException aoe && aoe.getAggregatedExceptions().size() == 1) {
+            exception = aoe.getAggregatedExceptions().values().iterator().next();
+        }
+        var now = ZonedDateTime.now(ZoneId.of("Z"));
         // @formatter:off
         return new KafkaProxyBuilder(primary)
                 .editOrNewStatus()
                     .withObservedGeneration(primary.getMetadata().getGeneration())
-                    .withConditions(effectiveReadyCondition(primary, exception))
+                    .withConditions(effectiveReadyCondition(primary, exception, now))
+                    .withClusters(clusterConditions(primary, context, now))
                 .endStatus()
             .build();
         // @formatter:on
     }
 
+    private static List<io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.Clusters> clusterConditions(KafkaProxy primary,
+                                                                                                             Context<KafkaProxy> context,
+                                                                                                             ZonedDateTime now) {
+        if (primary.getSpec() == null
+                || primary.getSpec().getClusters() == null) {
+            return List.of();
+        }
+        return primary.getSpec().getClusters().stream().map(cluster -> {
+            ClusterCondition clusterCondition = SharedKafkaProxyContext.clusterCondition(context, cluster);
+            var conditions = newClusterCondition(primary, clusterCondition, now);
+            return new io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.ClustersBuilder()
+                    .withName(cluster.getName())
+                    .withConditions(conditions).build();
+        }).toList();
+    }
+
     /**
      * Determines whether the {@code Ready} condition has had a state transition,
      * and returns an appropriate new {@code Ready} condition.
+     *
      * @param primary The primary.
      * @param exception An exception, or null if the reconciliation was successful.
+     * @param now
      * @return The {@code Ready} condition to use in {@code status.conditions}.
      */
     private static Conditions effectiveReadyCondition(KafkaProxy primary,
-                                                      @Nullable Exception exception) {
+                                                      @Nullable Exception exception,
+                                                      ZonedDateTime now) {
```

Comment:
I think there are two invariants we want:

1. If the cause of a condition with a given type (i.e. the status and reason) hasn't changed between one reconciliation and the next then its `lastTransitionTime` should not change. 
2. If a CR has two conditions in its status and they are both changed by a single reconciliation then we want them to have the same `lastTransitionTime` (and `observedGeneration`).

For point 2 that means we need to observe `now` once when building the status and use the same value for all conditions. Passing as an arg is the simplest way. I guess another option might be to set it as state in the `Context`. We might also consider abstracting away the call to get the system time too at some point. That would make our tests more deterministic.

As mentioned in other comments, I'm intentionally skimping on some of the testing of this stuff in this PR because in the next PR we'll have two different statuses (in different kinds) and there should be more opportunity to abstract this into some common handling. I'd rather wait and add tests for these semantics once I only have to write that prod code and test code once, rather than put in lots of work in this and the next PR.